### PR TITLE
Edge case for some encoders

### PR DESCRIPTION
**Pull request by borismcfishtank**
Changed start_pts and start_time to long to avoid overflow in json convertor.

### CODE REVIEW

**Comment by bramp** [comment]
Ah the tests have failed with 'com.google.gson.JsonSyntaxException: java.lang.NumberFormatException: Expected a long but was 0.040000 at line 33 column 37 path $.streams[0].start_time'

Start time should remain a double, as it is measured in seconds.

**Comment by bramp** [comment]
Oh and to be clear, start_time should be a double, but start_pts should be a long, since it is always a integer number.

**Comment by borismcfishtank** [comment]
Looks like I got carried away with the change to a long. I will rectify.

**Comment by bramp** [comment]
And if it is not too much hassle, adding a test case that ensures this stays a long, would be useful.

File: src/main/java/net/bramp/ffmpeg/probe/FFmpegStream.java [hunk 1]
```diff
@@ -44,7 +44,8 @@
   public Fraction avg_frame_rate;
   public Fraction time_base;
 
-  public int start_pts;
+  // Edge case found on encoder, changed from int to long
```
Comment (borismcfishtank) [inline src/main/java/net/bramp/ffmpeg/probe/FFmpegStream.java]: Quite true.

**Comment by borismcfishtank** [comment]
Couldn't think of a better place so I added the test to FFprobeTest.

File: src/test/java/net/bramp/ffmpeg/FFprobeTest.java [hunk 2]
```diff
@@ -57,6 +58,13 @@ public void testVersion() throws Exception {
   }
 
   @Test
+  public void testStartPtsType()
```
Comment (bramp) [inline src/test/java/net/bramp/ffmpeg/FFprobeTest.java]: This isn't the way I suspected you would check pts was a long. This tests how the code looks, now how it actually works.

Instead, I suggest updating 'fixtures/ffprobe-Always On My Mind [Program Only] - Adelén.mp4', to have a start_pts that is longer than a int can handle. Then test for the correct value in say FFprobeTest .testProbeVideo2.

Or, even better, take a ffprobe of your test file that exposed this problem in the first place, and add a new fixture, and new test case.

Comment (borismcfishtank) [inline src/test/java/net/bramp/ffmpeg/FFprobeTest.java]: Sure, seems like a much more complete way than my simple 'make sure no one has changed this from a long'.

Comment (bramp) [inline src/test/java/net/bramp/ffmpeg/FFprobeTest.java]: Unless we need the real file, just use the fake fixture. It'll keep the repo file size down, and avoids copyright issues :-)

Appreciate the effort

**Comment by borismcfishtank** [comment]
Except that the filename referenced in the samples class is different from the filename in the ffprobe results and that irritates me that I forgot to change that when I went back to the full size file for the ffprobe output.

**Comment by bramp** [comment]
Awesome. A lot of work for a simple change, but I think we are better off now! Thanks.

**Comment by borismcfishtank** [comment]
Thanks for reviewing my many erronious commits!

**Pull request merged**